Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -654,6 +662,34 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
         }
         return try description.symbolGraphExtractArguments()
     }
+
+    /// Returns the files and directories that affect the build process of this build plan.
+    package var inputs: [Input] {
```

Comment:
I'm not sure whether this is relevant but - traits are currently not considered by package structure command, is it possible for them to affect what files are included in the build and subsequently influence `fileAffectsSwiftOrClangBuildSettings`?

Reply:
cc @FranzBusch

Reply:
Traits aren't affecting what files are included in a single module. The same files will be part of the module just different `SWIFT_ACTIVE_COMPILATION_CONDITIONS` are passed. Traits do effect build planning though. The `swift build/run/test` commands support trait configuration arguments to change the traits of the root manifests. If the root trait changes we have to re-plan the build. This is already implemented and the trait configuration is stored in the build description.

What are the `package structure` commands in question here?

Reply:
PackageStructure command determines whether anything changed in the package and would skip loading of modules graph, and rebuilding of the build plan if nothing changed if traits are not part of that consideration there is a possibility of a split-brain scenario if something started directly using `buildPackageStructure()` result as an indicator for a rebuild.